LibreHardwareMonitorLib/Hardware/Cpu/CpuLoad.cs

```diff
@@ -42,32 +44,92 @@ private static bool GetTimes(out long[] idle, out long[] total)
-        int perfSize = Marshal.SizeOf(typeof(Interop.NtDll.SYSTEM_PROCESSOR_PERFORMANCE_INFORMATION));
-        if (Interop.NtDll.NtQuerySystemInformation(Interop.NtDll.SYSTEM_INFORMATION_CLASS.SystemProcessorPerformanceInformation, perfInformation, perfInformation.Length * perfSize, out int perfReturn) != 0)
-        {
-            return false;
-        }
-
-        idle = new long[idleReturn / idleSize];
-        for (int i = 0; i < idle.Length; i++)
-        {
-            idle[i] = idleInformation[i].IdleTime;
-        }
-
-        total = new long[perfReturn / perfSize];
-        for (int i = 0; i < total.Length; i++)
+        if (Software.OperatingSystem.IsUnix)
         {
-            total[i] = perfInformation[i].KernelTime + perfInformation[i].UserTime;
+            List<long> idleList = new();
+            List<long> totalList = new();
+            // Get the information
+            string[] cpuInfo = File.ReadAllLines("/proc/stat");
+            // currently parse the OverAll CPU info
+            // cpu   1583083 737    452845   36226266 723316   63685 31896     0       0       0
+            // cpu0  397468  189    109728   9040007  191429   16939 14954     0       0       0
+            // 0=cpu 1=user  2=nice 3=system 4=idle   5=iowait 6=irq 7=softirq 8=steal 9=guest 10=guest_nice
+            foreach (string cpuinfo in cpuInfo.Where(s => s.StartsWith("cpu")).Where(s => s[3] != ' '))
+            {
+                string[] overall = cpuinfo.Split(new char[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+                int cpuid;
+                long _idle = 0;
+                long user = 0, nice = 0, system = 0, iowait = 0;
+                long irq = 0, softirq = 0;
+                long steal = 0, guest = 0, guest_nice = 0;
+                try
+                {
+                    cpuid = Convert.ToInt32(overall[0][3]);
+                }
+                catch
+                {
+                    // ignore
+                }
+                // Parse idle information
+                try
+                {
+                    _idle = Convert.ToInt64(overall[4]);
+                }
+                catch
+                {
+                    _idle = 0;
+                }
+                idleList.Add(_idle);
+                // Parse total information
+                try
+                {
+                    user = Convert.ToInt64(overall[1]);
+                    nice = Convert.ToInt64(overall[2]);
+                    system = Convert.ToInt64(overall[3]);
+                    iowait = Convert.ToInt64(overall[5]);
+                    irq = Convert.ToInt64(overall[6]);
+                    softirq = Convert.ToInt64(overall[7]);
+                    steal = Convert.ToInt64(overall[8]);
+                    guest = Convert.ToInt64(overall[9]);
+                    guest_nice = Convert.ToInt64(overall[10]);
+                }
+                catch (Exception)
```

Comment:
This one too. I also think that it should not add on exception here, I'm not sure how useful it would be if you only have a subset of values.

Reply:
based on the version of linux kernel, several values might not be available.
the total of the load is taken from from the sum of "available" value..